src/effects/native/echoeffect.cpp

```diff
@@ -6,6 +6,7 @@
 #include "util/math.h"
 
 #define INCREMENT_RING(index, increment, length) index = (index + increment) % length
+#define DECREMENT_RING(index, decrement, length) index = (index + length - decrement) % length
```

Comment:
Why are preprocessor macros used for this? IMO it makes it harder to read. I'd rather read the math directly in the code where it is used or make functions.